lib/fly.js

```diff
@@ -85,12 +85,18 @@ module.exports = Fly
  * @return                      The current Fly instance.
  */
 Fly.prototype.source = function (globs) {
-	globs = arrify(globs)
+	globs = flatten([].slice.call(arguments))
```

Comment:
Your example is correct because that's what I would expect with that code and the use of `...globs`. However in the `flyjs` code it only useds `function (globs) {}` instead of `function (...globs) {}`.

I'm intentionally pulling it out because if it's not an object then it gets pushed back onto globs. But I just realized that I should be more specific in my check because `typeof []` returns `'object'`. 

I chose to use `.pop` instead of `globs.length - 1` because it's shorter. So instead of checking to see if the last value in `globs` is a plain object I'm check to see if it's not and then if it's not I'm pushing it back onto `globs`.


Below is what I can go with to have a more specific check for an plain object

```js
Fly.prototype.source = function (/*...*/globs, options) {
	globs = flatten([].slice.call(arguments))
	options = globs.pop()
	if (options.toString() !== '[object Object]') {
		globs.push(options)
		options = {}
	}
        ...
}
```


This allows you to use `this.source` like the following

```js
const globsofstuff = [
  'globofsomething/**/*',
  'globofsomethingelse/**/*',
]
export default function * () {
  await this.source('glob-1', 'glob-2') ...
  await this.source('glob-1', globsofstuff)... 
  await this.source('glob-1', globsofstuff, { ignore: 'someawesomepath' })... 
}
```